### PR TITLE
Refactor/vote registration store root canister id support

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

A refactoring to make `voteRegistrationStore` indifferent to neuron and proposal types.

# Changes

- `voteRegistrationStore` changes:
  - data by UniverseCanisterId 
  - `proposalInfo: ProposalInfo` -> `proposalIdSting: string`
  - `neuronIds: NeuronId[]` ->  `neuronIdStrings: string[]`
  - `successfullyVotedNeuronIds: NeuronId[]` -> successfullyVotedNeuronIdStrings: string[]`
- refactor `vote-registration.services` functions to make as much sns voting capable as possible
- `proposalsStore.reset()` for testing purposes
- fix and enable skipped voting tests

# Tests

- adjusted to the new data structure
- store supports universes

# Screenshot

Still works as expected!

Screen Recording 2023-04-12 at 17 39 01

### CODE REVIEW

File: frontend/src/lib/stores/vote-registration.store.ts [hunk 1]
```diff
@@ -1,126 +1,240 @@
...
 
-// TODO: think about having an abstract store for the basic CRUD
+export type VoteRegistrationStoreAddData = {
+  canisterId: UniverseCanisterId;
+} & Omit<
+  VoteRegistrationStoreEntry,
+  "status" | "successfullyVotedNeuronIdStrings"
+>;
+
+export type VoteRegistrationStoreAddSuccessfullyVotedNeuronIdData = {
```
Comment (lmuntaner) [inline frontend/src/lib/stores/vote-registration.store.ts:38]: 😂 

similar to pneumonoultramicroscopicsilicovolcanoconiosis

https://www.berlitz.com/blog/longest-word-english

File: frontend/src/tests/mocks/proposal.mock.ts [hunk 2]
```diff
@@ -72,9 +72,9 @@ export const mockProposalInfo: ProposalInfo = {
 } as unknown as ProposalInfo;
 
 export const mockVoteRegistration = {
-  proposalInfo: { ...mockProposalInfo },
-  neuronIds: [BigInt(0)],
-  successfullyVotedNeuronIds: [],
+  proposalIdString: "10000",
+  neuronIdStrings: ["0"],
```
Comment (lmuntaner) [inline frontend/src/tests/mocks/proposal.mock.ts:76]: Maybe use an id that is not 0 to not run into issues with 0 being falsy in tests?

File: frontend/src/lib/utils/neuron.utils.ts [hunk 3]
```diff
@@ -798,15 +799,15 @@ export const updateNeuronsVote = ({
 /** Is a neuron currently in a vote registration process */
 export const neuronVoting = ({
```
Comment (lmuntaner) [inline frontend/src/lib/utils/neuron.utils.ts:800]: Will it make sense to create a generic `neuronVoting` which expects also a root canister id, and then ths nnsNeuronVoting which uses OWN_CANISTER_ID?

**Review by lmuntaner — APPROVED** [review]
Looks good to me, sorry for the delay!

I added some questions, but approving already!

File: frontend/src/lib/stores/vote-registration.store.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/stores/vote-registration.store.ts:38]: Consistency is the way 🤓

File: frontend/src/tests/mocks/proposal.mock.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/mocks/proposal.mock.ts]: Makes sense. Done, thanks!

File: frontend/src/lib/utils/neuron.utils.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/lib/utils/neuron.utils.ts:800]: Thank you for the comment, I'll think about this in the upcoming PR. I guess there will be a clearer perspective when the sns voting is implemented. Soon in upcoming PR.

**Pull request merged**